packages/material-ui-lab/src/TreeItem/TreeItem.test.js

```diff
@@ -1473,14 +1495,513 @@ describe('<TreeItem />', () => {
+              <TreeItem nodeId="one" label="one" data-testid="one" />
+              <TreeItem nodeId="two" label="two" data-testid="two" />
+              <TreeItem nodeId="three" label="three" disabled data-testid="three" />
+              <TreeItem nodeId="four" label="four" data-testid="four" />
+              <TreeItem nodeId="five" label="five" data-testid="five" />
+            </TreeView>,
+          );
+
+          act(() => {
+            getByRole('tree').focus();
+          });
+          expect(getByTestId('one')).toHaveVirtualFocus();
+          fireEvent.keyDown(getByRole('tree'), {
+            key: 'End',
+            shiftKey: true,
+            ctrlKey: true,
+          });
+
+          expect(getByTestId('one')).to.have.attribute('aria-selected', 'true');
+          expect(getByTestId('two')).to.have.attribute('aria-selected', 'true');
+          expect(getByTestId('three')).to.have.attribute('aria-selected', 'false');
+          expect(getByTestId('four')).to.have.attribute('aria-selected', 'true');
+          expect(getByTestId('five')).to.have.attribute('aria-selected', 'true');
+        });
+
+        it('should prevent selection by keyboard home', () => {
+          const { getByRole, getByTestId } = render(
+            <TreeView multiSelect>
+              <TreeItem nodeId="one" label="one" data-testid="one" />
+              <TreeItem nodeId="two" label="two" data-testid="two" />
+              <TreeItem nodeId="three" label="three" disabled data-testid="three" />
+              <TreeItem nodeId="four" label="four" data-testid="four" />
+              <TreeItem nodeId="five" label="five" data-testid="five" />
+            </TreeView>,
+          );
+
+          fireEvent.focusIn(getByTestId('five'));
+          expect(getByTestId('five')).toHaveVirtualFocus();
+          fireEvent.keyDown(getByRole('tree'), {
+            key: 'Home',
+            shiftKey: true,
+            ctrlKey: true,
+          });
+
+          expect(getByTestId('one')).to.have.attribute('aria-selected', 'true');
+          expect(getByTestId('two')).to.have.attribute('aria-selected', 'true');
+          expect(getByTestId('three')).to.have.attribute('aria-selected', 'false');
+          expect(getByTestId('four')).to.have.attribute('aria-selected', 'true');
+          expect(getByTestId('five')).to.have.attribute('aria-selected', 'true');
+        });
+      });
+    });
+
+    describe('focus', () => {
+      describe('`disabledItemsFocusable=true`', () => {
+        it('should prevent focus by mouse', () => {
+          const focusSpy = spy();
+          const { getByText } = render(
+            <TreeView disabledItemsFocusable onNodeFocus={focusSpy}>
+              <TreeItem nodeId="one" label="one" data-testid="one" />
+              <TreeItem nodeId="two" label="two" disabled data-testid="two" />
+            </TreeView>,
+          );
+
+          fireEvent.click(getByText('two'));
+          expect(focusSpy.callCount).to.equal(0);
```

Comment:
Needed to be tested via the event handler due to the way events are simulated (fireEvent.click then tree.focus)